Microsoft.StorageSync 2025-12-01 New ApiVersion . Add changeEnumerationIntervalDays property to CloudEndpoint models

ankushbindlish2 opened the pull request:
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

<a href="?expand=1&template=sdk_configuration_template.md">Click here</a> to open a PR for only SDK configuration.

ankushbindlish2 committed:
Add new API version v2025_12_01

ankushbindlish2 committed:
Add StorageSync 2025-12-01 API examples

Adds comprehensive example JSON files for all CloudEndpoints, LocationOperationStatus, OperationStatus, PrivateEndpointConnections, RegisteredServers, ServerEndpoints, StorageSyncServices, SyncGroups, and Workflows operations for the stable 2025-12-01 API version. These examples provide maximum and minimum parameter sets for each operation, supporting documentation and testing.

ankushbindlish2 committed:
Add 2025-12-01 API version as default tag in readme.md

ankushbindlish2 committed:
Replace invalid auto-generated examples with validated 2022-09-01 examples for 2025-12-01 API version

ankushbindlish2 committed:
Add generated examples for 2025-12-01 API version with proper formatting

ankushbindlish2 committed:
Add StorageSync API version 2025-12-01 specification and examples

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Remove unreferenced example files from StorageSync 2025-12-01

ankushbindlish2 committed:
Merge branch 'storagesync-new-api-version-12-2025' of https://github.com/ankushbindlish2/azure-rest-api-specs into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Fix example files with proper JSON structure and api-version parameters

ankushbindlish2 committed:
Generate StorageSync API v2025-12-01 examples with proper ARM resource IDs

ankushbindlish2 committed:
Fix all resource ID fields in examples to use proper ARM resource ID format

ankushbindlish2 committed:
Fix storageAccountTenantId and offlineDataTransferStorageAccountTenantId to use proper UUID format

ankushbindlish2 committed:
Fix JSON parsing errors: remove trailing commas and invalid requestBody properties

ankushbindlish2 committed:
Fix trailing commas in PrivateEndpointConnections example files

ankushbindlish2 committed:
Format all example JSON files with prettier

ankushbindlish2 committed:
Add missing locationName parameter to StorageSyncServices_CheckNameAvailability_MinimumSet example

ankushbindlish2 committed:
Update StorageSync example responses and IDs

Replaces placeholder values in StorageSync example files with realistic resource IDs, names, and response headers. Adds Azure-AsyncOperation, Location, and Retry-After headers to 202 responses, and updates operation status and error details for consistency with actual API responses.

ankushbindlish2 committed:
Update StorageSync 2025-12-01 minimum set examples

Adds required parameters and response headers to all StorageSync minimum set example files for the 2025-12-01 API version. These changes improve example completeness and consistency, including operationId, subscriptionId, resourceGroupName, and other relevant fields, as well as more detailed response objects and headers.

ankushbindlish2 committed:
Add StorageSync 2025-12-01 API examples

Introduces new example JSON files for the 2025-12-01 API version under both 'examples' and 'stable' directories for Microsoft.StorageSync. These examples cover CloudEndpoints, PrivateEndpointConnections, RegisteredServers, ServerEndpoints, StorageSyncServices, SyncGroups, Workflows, and related operations, supporting documentation and testing for the updated API version. Also updates storagesync.json to reference the new examples.

ankushbindlish2 committed:
Add PATCH support and changeEnumerationIntervalDays to CloudEndpoint

Introduces a PATCH (update) operation for CloudEndpoint resources in the StorageSync API version 2025-12-01. Adds the changeEnumerationIntervalDays property to CloudEndpoint models, allowing configuration of the interval for enumerating changes, with validation for values between 1 and 20 days.

ankushbindlish2 committed:
Add CloudEndpoints update example files

Added maximum and minimum set example JSON files for the CloudEndpoints_Update operation in the 2025-12-01 API version. These examples provide reference payloads for updating cloud endpoints with both full and minimal parameter sets.

ankushbindlish2 committed:
Add CloudEndpoints_Update operation with changeEnumerationIntervalDays support for 2025-12-01 API version

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Update CloudEndpoint.tsp

ankushbindlish2 committed:
Merge branch 'storagesync-new-api-version-12-2025' of https://github.com/ankushbindlish2/azure-rest-api-specs into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Update storagesync.json

ankushbindlish2 committed:
Update storagesync.json

ankushbindlish2 committed:
Update CloudEndpoints_Update example references

Replaces separate minimum and maximum set examples with a unified CloudEndpoints_Update example. Updates example references in storagesync.json and removes the obsolete CloudEndpoints_Update_MinimumSet_Gen.json file.

ankushbindlish2 committed:
Mark CloudEndpoint provisioningState as read-only

Set the provisioningState property of CloudEndpoint to read-only in both 2022-09-01 and 2025-12-01 API versions. Updated TypeSpec and JSON schema, added client flattening for CloudEndpointUpdateProperties, and included new example files for CloudEndpoints_Update operations.

ankushbindlish2 committed:
Remove readOnly from CloudEndpoint provisioningState

The provisioningState property in CloudEndpoint is no longer marked as readOnly in both the TypeSpec and JSON schema files for 2022-09-01 and 2025-12-01. Updated tspconfig.yaml to use the read-only status schema option for consistency.

ankushbindlish2 committed:
Update ARM common types versioning and add suppressions

Moved @armCommonTypesVersion annotations to specific API versions in main.tsp for more granular control. Added suppressions for ProvisioningStateMustBeReadOnly and LroErrorContent in readme.md to address known issues with linting and error model consistency.

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Update main.tsp

ankushbindlish2 committed:
Merge branch 'storagesync-new-api-version-12-2025' of https://github.com/ankushbindlish2/azure-rest-api-specs into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ArthurMa1978 commented on specification/storagesync/resource-manager/Microsoft.StorageSync/StorageSync/CloudEndpoint.tsp:
```diff
@@ -74,6 +76,37 @@ interface CloudEndpoints {
+      requestId?: string;
+    }),
+    Error = StorageSyncError,
+    OptionalRequestBody = true
```
For `PATCH` operations, the request body is generally required. Please make sure whether setting `OptionalRequestBody = true` is appropriate.

ArthurMa1978 commented on specification/storagesync/resource-manager/Microsoft.StorageSync/StorageSync/back-compatible.tsp:
```diff
@@ -38,6 +38,11 @@ using Microsoft.StorageSync;
 
+#suppress "@azure-tools/typespec-azure-core/no-legacy-usage" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+#suppress "@azure-tools/typespec-azure-core/no-legacy-usage" "Back compatibility"
+@@Azure.ClientGenerator.Core.Legacy.flattenProperty(CloudEndpointUpdateParameters.properties
```
This is only kept for backward compatibility when converting the spec from Swagger to TypeSpec. For any newly added specs, do not use flatten anymore.

ArthurMa1978 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Merge branch 'Azure:main' into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Update CloudEndpoint patch requirements and schema details

Removed implicit optionality from CloudEndpoint patch operation and set request body as required. Cleaned up legacy suppressions and flattening in back-compatible.tsp. Updated OpenAPI schema to require 'properties' in CloudEndpointUpdateParameters, added 'uuid' format to relevant path parameters, and removed unused 'final-state-schema' and client flattening options.

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ArthurMa1978 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ArthurMa1978 committed:
update

sandipsh reviewed (approved):
:shipit:

mikeharder committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

Commit:
Add new sync activity fields

Commit:
Add sync activity fields to TypeSpec models

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

Commit:
Update example source files

ankushbindlish2 committed:
Merge pull request #6 from adamjseitz/storagesync-new-api-version-12-2025-sync-activity-fields

ankushbindlish2 commented:
> i have approved it with ArmSignedOff however the title says - "DoNotMerge", please explain what's the plan with this PR exactly.

Once the payload is deployed, We are going to remove it

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

ankushbindlish2 committed:
Apply tsp format to fix formatting

ankushbindlish2 committed:
update

ankushbindlish2 committed:
Merge branch 'main' into storagesync-new-api-version-12-2025

anpint commented:
Our RP has been updated to support this API version and it's enabled in the manifest so we're ready to publish it to customers

anpint commented:
@sandipsh @ArthurMa1978 Are you able to help get the PR merged?
